OrbitQt/main.cpp

```diff
@@ -239,9 +243,19 @@ static std::optional<OrbitQt::DeploymentConfiguration> FigureOutDeploymentConfig
   const char* const kEnvSignaturePath = "ORBIT_COLLECTOR_SIGNATURE_PATH";
   const char* const kEnvNoDeployment = "ORBIT_COLLECTOR_NO_DEPLOYMENT";
 
-  if (env.contains(kEnvExecutablePath) && env.contains(kEnvRootPassword)) {
+  // Program argument have higher precedence than environment variables.
+  std::string collector_override = absl::GetFlag(FLAGS_collector);
+  std::string collector_password = absl::GetFlag(FLAGS_collector_root_password);
+  std::string password = !collector_password.empty() ? collector_password
+                         : env.contains(kEnvRootPassword)
+                             ? env.value(kEnvRootPassword).toStdString()
+                             : "";
+
+  if (!collector_override.empty() && !password.empty()) {
+    return OrbitQt::BareExecutableAndRootPasswordDeployment{collector_override, password};
+  } else if (env.contains(kEnvExecutablePath) && !password.empty()) {
```

Comment:
Why not do the same for both location and password?
```suggestion
// Program argument have higher precedence than environment variables.
  std::string collector_location =
      !absl::GetFlag(FLAGS_collector).empty()
          ? absl::GetFlag(FLAGS_collector)
          : env.contains(kEnvExecutablePath) ? env.value(kEnvExecutablePath).toStdString() : "";
  std::string collector_password =
      !absl::GetFlag(FLAGS_collector_root_password).empty()
          ? absl::GetFlag(FLAGS_collector_root_password)
          : env.contains(kEnvRootPassword) ? env.value(kEnvRootPassword).toStdString() : "";

  if (!collector_location.empty() && !collector_password.empty()) {
    return OrbitQt::BareExecutableAndRootPasswordDeployment{collector_location, collector_password};
```

Reply:
Also.. should we support empty passwords? Probably not

Reply:
Good points @antonrohr , I will clean up the code and add support for empty password, thanks.